### PR TITLE
CIRC-730 Update check-in logic for actual cost fee refund

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
- update refund service to support actual cost fee for check-in;
- cover the scenario by tests;

Resolves: [CIRC-730](https://issues.folio.org/browse/CIRC-730)

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/FeeFine.java [hunk 1]
```diff
@@ -11,9 +11,11 @@ public class FeeFine {
   public static final String OVERDUE_FINE_TYPE = "Overdue fine";
   public static final String LOST_ITEM_FEE_TYPE = "Lost item fee";
   public static final String LOST_ITEM_PROCESSING_FEE_TYPE = "Lost item processing fee";
+  public static final String LOST_ITEM_FEE_ACTUAL_COST_FEE_TYPE = "Lost item fee (actual cost)";
```
Comment (roman-barannyk) [inline src/main/java/org/folio/circulation/domain/FeeFine.java:14]:
```suggestion
  public static final String LOST_ITEM_ACTUAL_COST_FEE_TYPE = "Lost item fee (actual cost)";
```

File: src/test/java/api/loans/scenarios/CheckInDeclaredLostItemTest.java [hunk 2]
```diff
@@ -137,4 +141,33 @@ void lostFeeCancellationDoesNotTriggerMarkingItemAsLostAndPaid() {
     assertThat(itemsClient.getById(item.getId()).getJson(), isAvailable());
   }
 
+  @Test
+  void shouldCancelItemFeeActualCostAndProcessingFee() {
+    final double itemFeeActualCost = 15.00;
+    final double itemProcessingFee = 10.00;
+    final UUID servicePointId = servicePointsFixture.cd1().getId();
+    UserResource user = usersFixture.charlotte();
+    feeFineOwnerFixture.ownerForServicePoint(servicePointsFixture.cd1().getId());
```
Comment (Mykyta-Varenyk) [inline src/test/java/api/loans/scenarios/CheckInDeclaredLostItemTest.java:150]:
```suggestion
    feeFineOwnerFixture.ownerForServicePoint(servicePointId);
```

Comment (roman-barannyk) [reply src/test/java/api/loans/scenarios/CheckInDeclaredLostItemTest.java]: fixed, thanks

File: src/test/java/api/loans/scenarios/CheckInDeclaredLostItemTest.java [hunk 3]
```diff
@@ -137,4 +141,33 @@ void lostFeeCancellationDoesNotTriggerMarkingItemAsLostAndPaid() {
...
+    final double itemFeeActualCost = 15.00;
+    final double itemProcessingFee = 10.00;
+    final UUID servicePointId = servicePointsFixture.cd1().getId();
+    UserResource user = usersFixture.charlotte();
+    feeFineOwnerFixture.ownerForServicePoint(servicePointId);
+    useLostItemPolicy(lostItemFeePoliciesFixture.create(
+      lostItemFeePoliciesFixture.facultyStandardPolicy()
+        .withName("Lost item fee actual cost policy")
+        .chargeProcessingFeeWhenDeclaredLost(itemProcessingFee)
+        .withActualCost(itemFeeActualCost)).getId());
```
Comment (Mykyta-Varenyk) [inline src/test/java/api/loans/scenarios/CheckInDeclaredLostItemTest.java:155]: Can you please try reusing Yusuf's method from:https://github.com/folio-org/mod-circulation/pull/1141/files#diff-6bf88f66480f263533b5d35a97c1f8b348d6f563d592bd535d4d3073e2358209

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**